lib/view_model/lightning_invoice_page_view_model.dart

```diff
@@ -0,0 +1,188 @@
+import 'dart:math';
+
+import 'package:cake_wallet/core/execution_state.dart';
+import 'package:cake_wallet/core/fiat_conversion_service.dart';
+import 'package:cake_wallet/entities/fiat_currency.dart';
+import 'package:cake_wallet/lightning/lightning.dart';
+import 'package:cake_wallet/store/settings_store.dart';
+import 'package:cake_wallet/view_model/lightning_view_model.dart';
+import 'package:cw_core/crypto_currency.dart';
+import 'package:cw_core/currency.dart';
+import 'package:cw_core/receive_page_option.dart';
+import 'package:cw_core/transaction_direction.dart';
+import 'package:cw_core/wallet_base.dart';
+import 'package:cw_core/wallet_type.dart';
+import 'package:cw_lightning/lightning_transaction_info.dart';
+import 'package:mobx/mobx.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+part 'lightning_invoice_page_view_model.g.dart';
+
+class LightningInvoicePageViewModel = LightningInvoicePageViewModelBase
+    with _$LightningInvoicePageViewModel;
+
+abstract class LightningInvoicePageViewModelBase with Store {
+  LightningInvoicePageViewModelBase(
+    this.settingsStore,
+    this._wallet,
+    this.sharedPreferences,
+    this.lightningViewModel, {
+    required this.useTorOnly,
+  })  : description = '',
+        amount = '',
+        state = InitialExecutionState(),
+        selectedCurrency = walletTypeToCryptoCurrency(_wallet.type),
+        cryptoCurrency = walletTypeToCryptoCurrency(_wallet.type) {
+    _fetchLimits();
+  }
+
+  List<Currency> get currencies => [walletTypeToCryptoCurrency(_wallet.type), ...FiatCurrency.all];
```

Comment:
do you take this into consideration when reading the invoice?

Reply:
I'm not sure what you mean by reading the invoice, but currently it's only used to construc the currency picker